sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample12_ConsumeEventsWithEventProcessor.cs

```diff
@@ -35,17 +40,250 @@ public class Sample12_ConsumeEventsWithEventProcessor : IEventHubsSample
         public async Task RunAsync(string connectionString,
                                    string eventHubName)
         {
-            // We will start by creating a client using its default set of options.
+            // We will start by creating a client using its default set of options.  It will be used by the event processor to
+            // communicate with the Azure Event Hubs service.
 
             await using (var client = new EventHubClient(connectionString, eventHubName))
             {
-                //TODO (caio, pri1): Populate the sample
+                // An event processor is associated with a specific Event Hub and a consumer group.  It receives events from
+                // all existing partitions in the Event Hub, passing them to the user for processing.
+                //
+                // Partition processors are associated with a specific partition and are responsible for processing events
+                // received from the event processor.  Their implementation must be provided by the user by creating a class
+                // that implements the IPartitionProcessor interface.
+                //
+                // An event processor must be able to create a partition processor for every partition it is responsible for.
```

Comment:
Maybe something like:  "The partition processor factory function is provided to the event processor when it is created.  The factory is responsible for creating a partition processor based on two arguments:   ...."